numpy/_core/getlimits.py

```diff
@@ -572,6 +577,33 @@ def _init(self, dtype):
         self._str_smallest_subnormal = machar._str_smallest_subnormal.strip()
         return self
 
+    def _init_from_finfo_obj(self, dtype, finfo_obj):
```

Comment:
Conceptually, this really is a class method, so I'd do
```
@classmethod
def _from_dtype_finfo(cls, dtype):
    self = object.__new__(cls):
    finfo = dtype.finfo()  # Would remove trailing "_obj".
    self.dtype = ...
```

Reply:
I agree with @mhvk here: constructor functions should be classmethods. 

@seberg I'm not sure what you're talking about here (even though I know mpmath)

Reply:
Sorry mindslip, thought this was somethig else :/